### DISCUSSION TITLE
Minimum OS or hardware requirements per platform to run exported game

### DISCUSSION THREAD

**Discussion by lostminds** [godotengine/godot-proposals#5946] [Platforms]
I've been looking around for a while now trying to find information about what the minimum requirements are to run my exported Godot game on various platforms. However, I've yet to find mostly other developers asking the same question in different forums, with only a few not so useful answers like "you've got to test that yourself" or "it depends on your game". 

While the latter is of course true when it comes to performance and "recommended specs", there must also be some minimum requirements on for example system OS version and maybe GPU hardware that come with the Godot engine itself, the renderer  you're using for the project and possibly other things. 

Like for example to run a Godot 4 forward_plus renderer game you need what version of macOS? linux? windows? Any minimum requirements on the CPU or GPU? Do the system requirements change if you use the mobile renderer? What android or iOS version is required to run the game?

So, first of all, is there already some resource somewhere that I've just not found with this minimum system requirements information, or at least some of it? 
If there is not, do you agree that this would be useful? 
If so, how could it be made available to developers working with games in Godot?
Maybe just a simple page with a table of requirements for various platforms and features in the godot docs? Or something more dynamic you could access from within the editor as part of the export process?

**Comment by Zireael07** [comment]
For Godot 4's Vulkan renderers (be it forward or mobile) the minimum is that your GPU is, well, Vulkan capable.

**Reply by lostminds** [reply]
Does that mean Vulkan 1.0 support or better then? If so I guess that's a good enough and simple answer regarding GPU requirements, for anyone looking this might then be a good more specific source [Vulkan support across vendors](https://en.wikipedia.org/wiki/Vulkan#Support_across_vendors)

**Reply by lostminds** [reply]
and since there's no support for Vulkan on Apple devices I guess it means any Metal capable GPU for macOS and iOS to support MoltenVK?

**Reply by Zireael07** [reply]
I'm not 100% certain what version of Vulkan Godot uses currently (it was changed a bit while in beta)

No clue about Apple devices.

Note that Godot 4 will have a GLES 3 renderer, that's why I noted _Vulkan_ renderer in my answer.

**Comment by Calinou** [comment]
Godot 4.0 is designed to work with full Vulkan 1.0 support, with optional usage of Vulkan 1.1 and 1.2 features. The partial Vulkan 1.0 support found in Intel Haswell IGPs on Linux is not considered sufficient (Mesa warns you on startup for this reason).

The OpenGL renderer in 4.0 requires OpenGL 3.3 Core Profile support on desktop platforms, OpenGL ES 3.0 on mobile platforms and WebGL 2.0 on HTML5.

CPU and memory requirements are highly variable depending on the project, but it's fair to say that on desktop platforms, you need to have a system with 4 GB of RAM at this point. 2 GB of RAM might work if the rest of your OS is highly optimized towards low RAM usage. On mobile platforms, 2 GB should be a safe bet, with 1 GB possibly working with the same caveat as desktop platforms. Using the OpenGL renderer instead of one of the Vulkan backends may help reduce memory usage (not tested).

For operating system versions, these are the current requirements for 4.0.beta as of December 2022:

- Windows 7 or later (32-bit or 64-bit x86).
- macOS 10.12 or later (64-bit, x86 or ARM ("Apple Silicon")). Godot supports Apple Silicon natively since 3.4 – no Rosetta is involved.
- Ubuntu 18.04 or later (32-bit or 64-bit x86, can be compiled for ARMv7 or ARMv8).
- Android 5.0 or later.
- iOS 12.0 or later (needs to be tested, it might be different).
- HTML5: a Web browser that supports WebAssembly with [SharedArrayBuffer](https://caniuse.com/sharedarraybuffer). This includes all up-to-date versions of common browsers, except Internet Explorer and Opera Mini.

**Reply by lostminds** [reply]
Thanks! This is all very good information, and the type I was looking for. 
Is this available somewhere in the docs for people to find?

**Reply by Calinou** [reply]
We don't have a set of system requirements that can be copy-pasted yet, but this can be added to the documentation. I am not sure what page this would be a good fit on though.

**Reply by lostminds** [reply]
Looking around for a good place in the docs I did actually find a list similar to yours above here: under "list of features" https://docs.godotengine.org/en/latest/about/list_of_features.html but I'm thinking that this is probably not an ideal place for it and since it's not updated I'm guessing it's sort of forgotten. 

So maybe move this information to a new page under "About" in the docs called "System requirements" or something? Adding a few more keywords to help people find it if they search the docs, and link to it from the export tutorials since that's a place where people might also be looking for the info?

**Reply by Calinou** [reply]
> and since it's not updated I'm guessing it's sort of forgotten.

I wrote that page back in mid-2020 and occasionally update it when I have time, but I'm usually busy with other pages. We also need testing reports to definitely confirm that Godot does run on the oldest specified operating system versions, as most contributors don't have access to devices running old OS versions (you usually can't downgrade after upgrading).

**Edit:** Issue opened: https://github.com/godotengine/godot-docs/issues/6497
